### DISCUSSION TITLE
Would adding all namespaces to useTranslation decrease performance on page-level?

### DISCUSSION THREAD

**Discussion by siigdev** [i18next/next-i18next#1708] [Q&A] [answered]
Hi,
I am having a single custom <Text> component that I am using to return all my translations. In here I have defined the useTranslation, and passing in the json translation key.  (I don't want to have the t in all my components, so I initialize it once).

On page level I have the serverSideTranslations with my specific namespaces.

Problem is that I can't use the custom Text component with useTranslation without defining the specific namespace. So my question is:
Would adding ALL possible namespaces to the custom Text component decrease performance, or does it only load the namespaces added in serverSideTranslations?

**Comment by isaachinman** [comment, marked as answer]
Yes, it would negatively affect performance. Each namespace you specify gets passed as props into the page-level component, via either `getStaticProps` or `getServerSideProps`.

The concept of namespaces is specifically to chunk out translation data to prevent the issue you are describing.

**Comment by siigdev** [comment]
Alright, thank you very much for your answer. That's unfortunately what I expected..

So in short, there is no good way to go around only initializing useTranslation once and just reuse that through props if there's multiple namespaces? I need to add useTranslation to all pages and components? Or am I not seeing something..
